src/main/java/ro/msg/learning/shop/service/OrderService.java

```diff
@@ -11,24 +11,36 @@
 import ro.msg.learning.shop.model.Stock;
 import ro.msg.learning.shop.repository.OrderRepository;
 import ro.msg.learning.shop.repository.StockRepository;
+import ro.msg.learning.shop.strategy.GreedyStrategy;
 import ro.msg.learning.shop.strategy.LocationStrategy;
+import ro.msg.learning.shop.util.RouteMatrixUtil;
 
+import java.math.BigDecimal;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 
 @Service
-@AllArgsConstructor
+@RequiredArgsConstructor
 public class OrderService {
 
     private final OrderRepository orderRepository;
     private final StockRepository stockRepository;
     private final LocationStrategy locationStrategy;
+    private final LocationService locationService;
+    private final RouteMatrixUtil routeMatrixUtil;
 
     @Transactional
     public Order createOrder(Order order, List<ProductQuantityDto> products) {
 
-        List<StockDto> stocksToBeOrdered = locationStrategy.findLocation(products);
+        List<BigDecimal> distances = null;
```

Comment:
This method should remain the same as it was before. You should not include any logic which is specific to any strategy. You should add the logic for GreedyStrategy inside the GreedyStrategy implementation.